BDArmory/Control/MissileFire.cs

```diff
@@ -3901,9 +3901,9 @@ void RefreshModules()
                 using (List<ModuleRadar>.Enumerator rd = _radars.GetEnumerator())
                     while (rd.MoveNext())
                     {
-                        if (rd.Current != null || rd.Current.canLock)
+                        if (rd.Current != null && rd.Current.canLock)
                         {
-                            if (rd.Current.maxLocks > MaxradarLocks) MaxradarLocks = rd.Current.maxLocks;
+                            if (rd.Current.maxLocks > 0) MaxradarLocks += rd.Current.maxLocks;
```

Comment:
You'll need to set MaxradarLocks to 0 before iterating over the radars, otherwise you'll just keep increasing the max number of locks.